test/tools/jsontests/StateTests.cpp

```diff
@@ -49,7 +49,7 @@ void doStateTests(json_spirit::mValue& _v, bool _fillin)
 		string testname = i.first;
 		json_spirit::mObject& o = i.second.get_obj();
 
-		if (_fillin)
+		if (_fillin && !Options::get().createRandomTest)
```

Comment:
stop being hilarious. we don't need 4 functions here. 
just one condition is being checked here:
```  if (mode is filler  and we are not in random test  )  then  (check that filename is the same as test name) ```